tools/gen_cli_usage.py

```diff
@@ -0,0 +1,45 @@
+import argparse
+import subprocess
+import sys
+
+
+def get_usage_lines() -> list:
+    res = subprocess.run(["uvicorn", "--help"], stdout=subprocess.PIPE)
+    return res.stdout.decode("utf-8").splitlines()
+
+
+def main(check: bool = False) -> None:
+    with open("docs/index.md") as f:
+        content = f.read()
+
+    lines = content.splitlines()
+    marker_lineno = lines.index("<!-- gen_cli_usage:marker -->")
+    assert lines[marker_lineno + 1] == "```"
+    start = marker_lineno + 2
+    end = next(
+        lineno for lineno, line in enumerate(lines[start:], start) if line == "```"
+    )
+    lines = lines[:start] + get_usage_lines() + lines[end:]
```

Comment:
Honestly, these kind of < 10 LOC scripts that do useful tasks yet remain pretty readable is why I love Python so much. ❤️